src/dataframe/dataframe.jl

```diff
@@ -1100,9 +1124,59 @@ function permutecols!(df::DataFrame, p::AbstractVector)
         throw(ArgumentError("$p is not a valid column permutation for this DataFrame"))
     end
     permute!(columns(df), p)
+    permute!(metadata(df), p)
     setfield!(df, :colindex, Index(names(df)[p]))
 end
 
 function permutecols!(df::DataFrame, p::AbstractVector{Symbol})
     permutecols!(df, getindex.(index(df).lookup, p))
 end
+
+
+##############################################################################
+##
+## Set and Get MetaData 
+##
+##############################################################################
+
+"""
+    addlabel!(df::DataFrame, var::Symbol, label::String)
+
+Adds a label to a DataFrame. Does not add other metadata.
+"""
+function addmeta!(df::DataFrame, var::Symbol, field::Symbol, info)
+    ind = index(df)[var]
+    # pass the number of columns to the function so that it can create a new array of 
```

Comment:
That sounds OK to me (and no need for a comment).

BTW, these functions would be turned into one-liners if you skip creating the `ind` variable.